frontend/public/components/storage/create-pvc.tsx

```diff
@@ -64,7 +80,8 @@ export class CreatePVCForm extends React.Component<CreatePVCFormProps, CreatePVC
   };
 
   handleStorageClass = storageClass => {
-    this.setState({ storageClass: _.get(storageClass, 'metadata.name'), storageClassAccessMode: _.get(storageClass, ['metadata', 'annotations', 'storage.alpha.openshift.io/access-mode'])}, this.onChange);
+    const {provisionerAccessModeMapping} = this.state;
+    this.setState({ storageClass: _.get(storageClass, 'metadata.name'), storageClassAccessModes: provisionerAccessModeMapping[storageClass.provisioner]}, this.onChange);
```

Comment:
Reading from state, modifying the value, then calling `setState` can be problematic. See

https://reactjs.org/docs/faq-state.html#why-is-setstate-giving-me-the-wrong-value

This problem goes away though if we move the map out of state, however.

Reply:
We should be careful that we don't have a runtime error if we ever see a storage class provisioner we don't recognize. I would make sure we default to an empty array if the access mode mappings don't have the key.

Reply:
Ack. Should we need to set any default access mode or no access modes will be enabled at all?